src/supported-operations/thin-edge-supported-operations.md

```diff
@@ -0,0 +1,127 @@
+# Supported operations
+
+## Support for operations
+
+Supported operations would be a part of thin-edge data model and supported by `operation configuration` files in the filesystem.
+
+Supported operations are listed per supported cloud.
+
+Supported operations can be modified using tedge cli tool.
+
+Operations supported by `thin-edge.io` would be stored in the `operations` directory of `thin-edge.io` configuration, it is `/etc/thin-edge/operations` by default.
+Single operation type is expressed by operation configuration file in the filesystem.
+
+Many `thin-edge.io` components can read operation configuration files and use them to perform operations, but only one component should report operations to an external connection (if supported by provider, eg. for c8y cloud it could be mapper).
+
+> Alternatives:
+
+### Option 1
+
+Operations use prefixes to identify the cloud, eg `c8y_SoftwareUpdate`, `az_SoftwareUpdate`.
+
+### Option 2
+
+Per cloud operation is achieved by use of subdirectories, eg `c8y/`, eg: `/etc/tedge/operations/c8y/c8y_SoftwareUpdate` (in this example `c8y_` in the name of operation is required as it is actual name of the operation in c8y, this could be avoided but more on this later), ...
+
+> Done
+
+The user can add operations by adding the operation configuration file to the filesystem at any time of `thin-edge.io` lifecycle.
+
+Configuration files are in toml format and can be empty files denoting no additional configuration or parameters are required (eg, natively supported operations or trivial operations).
+
+The configuration file name is the same as the operation name.
+
+> Consideration: FS permissions, what to set, who should be able to change it?
+
+## Operation configuration
+
+Operations may need to be configured to be executed as intended. For example `c8y_LogfileRequest` requires additional parameter to be set, `log_type` in this case. This can be achieved by adding a configuration settings in the configuration file following the proposed structure.
+
+All operation configuration files are `toml` format.
+
+Some tables are optional, but if present they must be in the following format:
+
+```toml
+[exec]
+  # Exec configuration if the operation requires command execution
+  # Required
+  command = "echo"
+  # Optional
+  args = ["hello"]
```

Comment:
How do you handle arguments that come from the cloud (e.g. remote access sends websocket URL)?

Reply:
Indeed, this is not clear.

I feel that the simpler is to let the operation plugin retrieve *itself* its parameters from the former MQTT message that triggered the operation as well as the sub-sequent ones. I.e the responsibility of thin-edge is to awake the plugin not to prepare the operation.

Reply:
I agree that making it static didn't make it flexible enough.
But leaving it out still doesn't solve the issue of passing commands from cloud to executors.
I would say for drop 1 this can be hardcoded, but for full flexibility I would propose some kind scheme which can be provided for the command to translate from cloud type to thin-edge type.